rms/search_engine/clients.py

```diff
@@ -0,0 +1,57 @@
+from abc import ABC
+from typing import Any
+
+import httpx
+from rms.search_engine.models import SearchBody, ScopusSearchResponse
+from rms.settings import Settings
+
+
+class ScopusClient(ABC):
+
+    def __init__(self):
+        self.settings = Settings()
+        self.HEADERS = {
+            "httpAccept": "application/json",
+            "X-ELS-APIKey": self.settings.scopus_api_key
+        }
+        self.SCOPUS_SEARCH_API_ENDPOINT = self.settings.scopus_search_endpoint
+
+
+class ScopusArticleSearchApi(ScopusClient):
+
+    @staticmethod
+    def build_params(query: SearchBody) -> dict[str, Any]:
+        search_criteria = []
+
+        if query.title:
+            title_parts = query.title.split(" ")
+            title_criteria = " OR ".join([f'TITLE("{part}")' for part in title_parts])
+            search_criteria.append(f"({title_criteria})")
+
+        if query.keywords:
+            keywords_criteria = " OR ".join([f'KEY("{keyword}")' for keyword in query.keywords])
+            search_criteria.append(f"({keywords_criteria})")
+
+        if query.abstract_keywords:
+            abstract_keywords_criteria = " OR ".join([f'ABS("{keyword}")' for keyword in query.abstract_keywords])
+            search_criteria.append(f"({abstract_keywords_criteria})")
```

Comment:
I would separate this into 3 private method, each of which would handle the check(`if query.title` for example) inside. Afterwards the main function body can get the search criteria in just 3 linear fn calls